### ISSUE TITLE
Hostname or IP as an output of domain

### ISSUE THREAD

**Issue by moio** [dmacvicar/terraform-provider-libvirt#89]
A domain's IP/hostname might not be known before the resource is created, for example if bridged networking is used with an external DHCP.

libvirt knows though - possibly through the qemu guest agent - can we make this an output of the domain resource for later use?

**Comment by inercia** [comment]
Not sure if this is what you mean, but Terraform knows about all the IP addresses in the cluster, so you can just use the `output` statement:

```
output "ip_salt" {
  value = "${libvirt_domain.salt.network_interface.0.addresses.0}"
}
```

**Comment by inercia** [comment]
> I can contribute a documentation PR, but how can I determine the list of all exported attributes?

Look at the `tfstate` and the `terraform state` command. In my example, `libvirt_domain.salt.network_interface.0.addresses.0` is just the path to a leaf in the JSON contained in the state file...

**Comment by moio** [comment]
Addressed in #90
